impl/src/main/java/org/ehcache/impl/internal/executor/PooledExecutionService.java

```diff
@@ -129,14 +130,32 @@ public void stop() {
       }
     }
     try {
-      if(!scheduledExecutor.awaitTermination(30, SECONDS)) {
-        LOGGER.warn("Timeout while waiting on scheduler to finish");
+      while(!scheduledExecutor.awaitTermination(30, SECONDS)) {
```

Comment:
That makes sense... and looking at the other executor shutdown paths seems to be consistent with existing behavior.